File: easy/binary-search/binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        
        left_bound = 0
        right_bound = len(nums) - 1
        while left_bound <= right_bound:
            midpoint = left_bound + (right_bound - left_bound) // 2
            if nums[midpoint] == target:
              return midpoint
            if nums[midpoint] > target:
              right_bound = midpoint - 1
            if nums[midpoint] < target:
              left_bound = midpoint + 1
        return -1
      

File: easy/binary-search/binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size();
        if (nums.size() == 1) {
            if (nums[0] == target) {
                return 0;
            }
            return -1;
        }
        while (l <= r) {
            int m = l + (r - l) / 2;
            if (nums[m] == target) {
                return m;
            }
            if (nums[m] > target) {
                r = m-1;
            }
            if (nums[m] < target) {
                l = m+1;
            }
        }
        return -1;
    }
};
